### DISCUSSION TITLE
TypeError: JSON.stringify cannot serialize cyclic structures.

### DISCUSSION THREAD

**Discussion by keisto** [alpinejs/alpine#1394] [7. Bugs (archived)]
I may not be doing this right, but I have this input that displays money in decimal format `100.00` but then use alpine to format into pennies `10000` for the database.

<img width="870" alt="Screen Shot 2021-03-24 at 7 05 52 PM" src="https://user-images.githubusercontent.com/2490903/112407800-f6b55880-8cd3-11eb-8471-787b82a70b66.png">

It works perfectly! However every time I hit save I get this error thrown, because entangle changes from `10000` to a Livewire object?

<img width="825" alt="Screen Shot 2021-03-24 at 7 06 20 PM" src="https://user-images.githubusercontent.com/2490903/112407860-07fe6500-8cd4-11eb-9aca-5f3ee4ff73eb.png">

**Comment by IP-Developer** [comment]
Can confirm. Got the same error with entangle.

**Comment by HugoDF** [comment]
Is this an Alpine issue?

Looks like the code causing this is in Livewire https://github.com/livewire/livewire/blob/master/js/component/SupportAlpine.js

**Comment by Ryuno-Ki** [comment]
Sounds like a use case for a library like [cycle.js](https://github.com/douglascrockford/JSON-js) that offers methods to decycle those references.
